docs/populate.md

```diff
@@ -72,21 +68,22 @@ const author = new Person({
   age: 50
 });
 
-author.save(function(err) {
-  if (err) return handleError(err);
-
-  const story1 = new Story({
-    title: 'Casino Royale',
-    author: author._id // assign the _id from the person
-  });
+await author.save();
 
-  story1.save(function(err) {
-    if (err) return handleError(err);
-    // that's it!
-  });
+const story1 = new Story({
+  title: 'Casino Royale',
+  author: author._id // assign the _id from the person
 });
+
+await author.save();
+// that's it!
 ```
 
+You can set the `ref` option on `ObjectId`, `Number`, `String`, and `Buffer` paths.
+`populate()` works with ObjectIds, numbers, strings, and buffers.
+However, we recommend using ObjectIds as `_id` properties (and thus ObjectIds for `ref` properties) unless you have a good reason not to.
+That is because MongoDB will set `_id` to an ObjectId if you create a new document without an `_id` property, so if you make your `_id` property a Number, you need to be extra careful not to insert a document without a numeric `_id`.
```

Comment:
maybe mention that mongoose can help with that via validators, like using `required: true` and for non-default types to have to use `default: () => newType` to generate a new id (via a function instead of statically)